### PR TITLE
Create and show zones

### PR DESCRIPTION
**Pull request by Hristo-Boyadzhiev**
https://github.com/orgs/InterCarsBulgaria-TechLeadAcademy/projects/3/views/1?pane=issue&itemId=70721498

### CODE REVIEW

File: src/hooks/services/zones/useGetZones.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { getWarehouseManagementApi } from '@/services/generated-api'
+import { useSuspenseQuery } from '@tanstack/react-query'
+
+export default function useGetZones() {
+  //TODO: use pagination, when back-end team done
```
Comment (vexleet) [inline src/hooks/services/zones/useGetZones.ts]: lets not have pagination for this one too

File: vite.config.ts.timestamp-1720594604817-0926b8ddf815b.mjs
Comment (vexleet) [inline vite.config.ts.timestamp-1720594604817-0926b8ddf815b.mjs:1]: delete this file

File: src/components/features/forms/NewZoneForm.tsx [hunk 2]
```diff
@@ -10,21 +12,39 @@ import {
...
 } from '@mui/material'
+import { useSuspenseQuery } from '@tanstack/react-query'
 import { Controller, UseFormReturn } from 'react-hook-form'
 import { useTranslation } from 'react-i18next'
 
-const markers = ['Масло', 'Гуми', 'Чистачки']
+interface NewZoneFormProps extends UseFormReturn<NewZoneFormData> {
+  defaultValues?: {
+    name?: string
+    markersIds?: any //трябва да е [] | string[], но така гърми ZoneTableActionsMenu и не знам как
```
Comment (vexleet) [inline src/components/features/forms/NewZoneForm.tsx]: should be `number[]`,from what i remember marker id is a `number`

File: src/components/features/actionsMenu/ZonesTableActionsMenu.tsx [hunk 3]
```diff
@@ -24,47 +30,60 @@ export default function ZonesTableActionsMenu() {
...
+          confirmText={translate('newZone.editZone.labels.edit')}
+          onCloseDialog={handleClose}
+          schema={newZoneSchema}
+          onSubmit={handleSubmit}
+          renderForm={(methods) => (
+            <NewZoneForm
+              {...methods}
+              defaultValues={{
+                name: zone.name!,
+                markersIds: zone.markers?.map((marker) => marker.markerId!) || ([] as string[]),
```
Comment (vexleet) [inline src/components/features/actionsMenu/ZonesTableActionsMenu.tsx]: should be `[] as number[]`

File: src/hooks/services/zones/useGetZones.ts (hunk 1 above)
Comment (Hristo-Boyadzhiev) [reply src/hooks/services/zones/useGetZones.ts]: Deleted comment

File: vite.config.ts.timestamp-1720594604817-0926b8ddf815b.mjs
Comment (Hristo-Boyadzhiev) [reply vite.config.ts.timestamp-1720594604817-0926b8ddf815b.mjs:1]: Deleted

File: src/components/features/actionsMenu/ZonesTableActionsMenu.tsx (hunk 3 above)
Comment (Hristo-Boyadzhiev) [reply src/components/features/actionsMenu/ZonesTableActionsMenu.tsx]: Fixed

File: src/components/features/forms/NewZoneForm.tsx (hunk 2 above)
Comment (Hristo-Boyadzhiev) [reply src/components/features/forms/NewZoneForm.tsx]: Fixed

**Review by vexleet — APPROVED** [review]

**Pull request merged**